test/controller_inheritance_test.rb

```diff
@@ -0,0 +1,54 @@
+require "test_helper"
+
+class PostsInheritanceTest < ActionDispatch::IntegrationTest
+  test "it works" do
+    post = Post.create!
+
+    get("/alt_posts")
+
+    json = JSON.parse(@response.body)
+    assert_equal 1, json.size
+    assert_equal(post.id, json.first["id"])
+  end
+
+  test "it inherits filter from parent controller" do
+    post = Post.create!
+    Post.create!
+
+    get("/alt_posts", params: { filters: { id: post.id } })
+
+    json = JSON.parse(@response.body)
+    assert_equal 1, json.size
+    assert_equal post.id, json.first["id"]
+  end
+
+  test "it inherits sort from parent controller" do
+    Post.create!(title: "z")
+    Post.create!(title: "a")
+
+    get("/alt_posts", params: { sort: "title" })
+
+    json = JSON.parse(@response.body, object_class: OpenStruct)
+    assert_equal ["a", "z"], json.map(&:title)
+  end
+
+  test "it overrides filter from parent controller" do
+    Post.create!(body: "hi")
+    Post.create!(body: "hello")
+    Post.create!(body: "hola")
+    get("/alt_posts", params: { filters: { french_bread: ["hi", "hello"] } })
```

Comment:
This feels like more of an issue with how the tests are setup, but it's not totally clear as to whether this is actually working as expected, since it's not obvious what the original implementation of the `french_bread` (not sure who originally chose that name lol) filter is. 

In lieu of refactoring the tests, a comment might be helpful here!

Reply:
Totally agree. Instead of overriding `french_bread` and `foobar`, I made `body` alias to `priority` for both filter and sort. I think it's much clearer now what is being overridden.